terraform/vpc.tf

```diff
@@ -0,0 +1,199 @@
+resource "aws_vpc" "main" {
+    cidr_block = var.vpc-cidr
+    enable_dns_hostnames = true
+    tags = { Name = "tfvpc"}
+}
+resource "aws_subnet" "public-subs" {
+    vpc_id = aws_vpc.main.id
+  count = length(var.publicsubnets)
+  cidr_block = var.publicsubnets[count.index].cidr
+  availability_zone = var.publicsubnets[count.index].az
+  map_public_ip_on_launch = var.publicsubnets[count.index].publicip
+  tags = {
+     Name = var.publicsubnets[count.index].name
```

Comment:
add the project name variable to identify better the subnets. Imagine if you have many subnets for multiple vpc all named as "public-*"

Name = "${var.project_name}-${var.var.publicsubnets[count.index].name}"

do the same for the other subnets